src/scripts/main.js

```diff
@@ -21,7 +21,25 @@ const food = {
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  if (!data || typeof data !== 'object') {
+    return null;
+  }
+
+  const ul = document.createElement('ul');
+
+  for (const key in data) {
+    const li = document.createElement('li');
+
+    li.textContent = key;
+
+    if (typeof data[key] === 'object') {
+      createTree(li, data[key]);
+    }
```

Comment:
The condition `if (typeof data[key] === 'object')` should also check if the object is not empty to avoid unnecessary recursive calls. Consider using `Object.keys(data[key]).length > 0` to ensure the object has properties before calling `createTree` recursively.